crates/nu-protocol/src/engine/cached_file.rs

```diff
@@ -0,0 +1,12 @@
+use crate::Span;
+use std::sync::Arc;
+
+#[derive(Clone)]
+pub struct CachedFile {
+    // Could possibly become an `Arc<PathBuf>`
+    pub name: Arc<str>,
+    // Use Arcs of slice types for more compact representation (capacity less)
+    pub content: Arc<[u8]>,
+    // TODO: when refactoring `Span` to IDs this needs to remain an interval
+    pub covered_span: Span,
```

Comment:
You mean to keep it as Span, not converting to SpanId?

Reply:
As long as we translate to `Span` with a global index we would need that to translate back to indices into `content`

Reply:
After looking how #12239 works right now I removed this particular comment, and documented the struct publicly.